Brokerages/Oanda/OandaBrokerage.cs

```diff
@@ -0,0 +1,1052 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net;
+using System.Runtime.Serialization.Json;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml;
+using Newtonsoft.Json;
+using QuantConnect.Brokerages.Oanda.DataType;
+using QuantConnect.Brokerages.Oanda.DataType.Communications;
+using QuantConnect.Brokerages.Oanda.Framework;
+using QuantConnect.Brokerages.Oanda.Session;
+using QuantConnect.Brokerages.Tradier;
+using QuantConnect.Logging;
+using QuantConnect.Orders;
+using QuantConnect.Securities;
+using RestSharp;
+using Order = QuantConnect.Orders.Order;
+
+namespace QuantConnect.Brokerages.Oanda
+{
+    /// <summary>
+    /// Oanda Brokerage model.
+    /// </summary>
+    public class OandaBrokerage : Brokerage
+    {
+        /// <summary>
+        /// The account identifier for Oanda Accounts.
+        /// </summary>
+        public int AccountId { get; private set; }
+
+        private DateTime _issuedAt;
+
+        private TimeSpan _lifeSpan = TimeSpan.FromSeconds(86399); // 1 second less than a day
+        
+
+        private readonly object _fillLock = new object();
+
+        /// <summary>
+        /// Gets or sets the instrument security type map.
+        /// </summary>
+        /// <value>
+        /// The instrument security type map.
+        /// </value>
+        public Dictionary<string, SecurityType> InstrumentSecurityTypeMap { get; private set; }
+
+        /// <summary>
+        /// Gets the oanda environment.
+        /// </summary>
+        /// <value>
+        /// The oanda environment.
+        /// </value>
+        public static Environment OandaEnvironment { get; private set; }
+
+        private readonly object _lockAccessCredentials = new object();
+
+        //This should correlate to the Trades/Positions list in Oanda.
+        private readonly IHoldingsProvider _holdingsProvider;
```

Comment:
this appears to be unused. in the tradier impl we needed this since if an order will bring your holdings over the zero boundary it needs to be split into two orders. for example, I own 10 shares SPY and sell 20, tradier requires a sell to close the 10 and then a sell to open. does oanda have a similar requirement?